### PR TITLE
Support for Windows cross-compilation from Linux

### PR DESCRIPTION
**Pull request by zah**
Since `DirSep` matches the default separator of the target OS, it's not suitable for performing path operations during compile-time on the host OS.

The particular fix here solves the issue for cross-compiling a Windows build from a Linux host, but Nim seems to need a more general solution for detecting the host OS during compilation, so a host OS specific separator can be used when deriving paths from `currentSourcePath`.

### CODE REVIEW

File: bindings/nim/kzg_abi.nim
Comment (asn-d6) [inline bindings/nim/kzg_abi.nim:13]: This was just changed here: https://github.com/ethereum/c-kzg-4844/pull/294

Before merging this I think the rest of the nimbus team should review this to make sure that it's final.

Comment (zah) [reply bindings/nim/kzg_abi.nim:13]: I'm a member of the Nimbus team, but this change was accidental. Will remove it from the PR.

Comment (zah) [reply bindings/nim/kzg_abi.nim]: The fix has been pushed.

File: bindings/nim/kzg_abi.nim
Comment (jtraglia) [inline bindings/nim/kzg_abi.nim:15]: I believe the `"/"` appended at the end is platform specific, right? Should it be replaced with `DirSep`, or in the case of cross-compiling for Windows `AltSep`?

Comment (jangko) [reply bindings/nim/kzg_abi.nim:15]: Both backward and forward slash will works on Windows, combination of both also works. The literal "/" will works for cross-compiling for Windows.
If we use `DirSep` or `AltSep` it will become a problem because Nim will translate it into host OS separator, not target  OS separator. IIRC there is a lengthy discussion and various reasons came up why adding cross compilation features to Nim compiler(especially about path handling) is difficult.

Comment (zah) [reply bindings/nim/kzg_abi.nim:15]: No, the current solution is best because it works correctly on both Linux and Windows, both locally and when cross-compiling. We rely on the fact that Windows supports `/` as a separator.

Comment (jtraglia) [reply bindings/nim/kzg_abi.nim:15]: Okay gotcha, thanks guys.

**Review by jtraglia — APPROVED** [review]
LGTM 👍

**Pull request merged**
